### PR TITLE
fix: payroll issues

### PR DESCRIPTION
**Pull request by AfshanKhan**
backport of https://github.com/frappe/erpnext/pull/24528 and https://github.com/frappe/erpnext/pull/24529

Also:
Before:
Screen Recording 2021-02-12 at 12 14 42 PM

After:
Screen Recording 2021-02-12 at 12 16 04 PM

### CODE REVIEW

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.js [hunk 1]
```diff
@@ -133,45 +133,55 @@ frappe.ui.form.on('Payroll Entry', {
...
 		});
 	},
 
 	company: function (frm) {
 		frm.events.clear_employee_table(frm);
 		erpnext.accounts.dimensions.update_dimension(frm, frm.doctype);
+		frm.trigger("fill_details");
+	},
+
+	fill_details: function (frm) {
```
Comment (Anurag810) [inline erpnext/payroll/doctype/payroll_entry/payroll_entry.js]: you can name it `set_payable_account_and _currency`, `fill_details` is little ammbigous.

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.js [hunk 2]
```diff
@@ -133,45 +133,55 @@ frappe.ui.form.on('Payroll Entry', {
...
+			}
+			if (frm.doc.branch) {
+				filters['branch'] = frm.doc.branch;
+			}
+			if (frm.doc.designation) {
+				filters['designation'] = frm.doc.designation;
+			}
+			if (frm.doc.employees) {
+				filters['employees'] = frm.doc.employees.filter(d => d.employee).map(d => d.employee);
+			}
```
Comment (Anurag810) [inline erpnext/payroll/doctype/payroll_entry/payroll_entry.js]: you can move this to function and name it `get_employee_filters` or `get_employee_query_condiitons`

File: erpnext/payroll/doctype/payroll_entry/payroll_entry.py [hunk 3]
```diff
@@ -563,25 +563,16 @@ def create_salary_slips_for_employees(employees, args, publish_progress=True):
...
 	payroll_entry = frappe.get_doc("Payroll Entry", args.payroll_entry)
 	payroll_entry.db_set("salary_slips_created", 1)
 	payroll_entry.notify_update()
 
+	if salary_slips_not_created:
+		frappe.msgprint(_("Salary Slips already exists for employees {}, and will not be processed by this payroll.")
+			.format(frappe.bold(", ".join([emp for emp in salary_slips_not_created]))) , title=_("Message"), indicator="orange")
+
 def get_existing_salary_slips(employees, args):
 	return frappe.db.sql_list("""
```
Comment (Anurag810) [inline erpnext/payroll/doctype/payroll_entry/payroll_entry.py:582]: please update this query and get salary slips based on payroll entry reference in the salary slips document.

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Review by Anurag810 — COMMENTED** [review]
@AfshanKhan Fix Sider. Waiting for green check...

**Comment by AfshanKhan** [comment]
> @AfshanKhan Fix Sider. Waiting for green check...

@Anurag810 Slider is fixed.

**Pull request merged**